testtools/matchers/_impl.py

```diff
@@ -117,11 +124,13 @@ def __init__(self, matchee, matcher, mismatch, verbose=False):
         super(MismatchError, self).__init__()
         self.matchee = matchee
         self.matcher = matcher
-        self.mismatch = mismatch
+        self.mismatch = IMismatch(mismatch, mismatch)
```

Comment:
So, I very much dislike this, as its referencing the global registry. That lets anyone break this code accidentally :(.